log/scope.go

```diff
@@ -387,12 +398,37 @@ func (s *Scope) WithLabels(kvlist ...interface{}) *Scope {
 	return out
 }
 
+func (s *Scope) WithRateSampling(limit rate.Limit, burst int) *Scope {
```

Comment:
Would be good to comment on here what happens if we hit the limit - does it stall, drop the write, async write later, etc

Reply:
Added a bit of commentary. please let me know what you think.